### PR TITLE
Helper for printing TERM_COLORS

### PR DESCRIPTION
**Pull request by nicfit**
Handles the \001 and \002 replacements.

### CODE REVIEW

File: xonsh/tools.py [hunk 1]
```diff
@@ -757,3 +757,7 @@ def format_prompt_for_prompt_toolkit(prompt):
     token_names = [get_xonsh_color_names(c) for c in parts[::2]]
     cstyles = [_make_style(c) for c in token_names]
     return token_names, cstyles, strings
+
+
+def print_color(string):
```
Comment (scopatz) [inline xonsh/tools.py]: This function needs a docstring.

File: xonsh/readline_shell.py [hunk 2]
```diff
@@ -204,9 +204,7 @@ def _cmdloop(self, intro=None):
                     if inserter is not None:
                         readline.set_pre_input_hook(None)
                 else:
-                    self.stdout.write(self.prompt.replace('\001', '')
-                                                 .replace('\002', ''))
-                    self.stdout.flush()
+                    print_color(self.prompt)
```
Comment (scopatz) [inline xonsh/readline_shell.py]: This does not exactly replicate the code that was there originally.  In particular, the original uses `self.stdout`, while the `print_color()` function will go to `sys.stdout`.

**Comment by scopatz** [comment]
Thanks for putting this in @nicfit

**Comment by nicfit** [comment]
It needs work, but it sounds like there is some interest so..I add a docs, and a file kwarg so self.stdout can be passed (or sys.stderr). I'll reopen then.

**Comment by scopatz** [comment]
Is this ready @nicfit?

**Comment by nicfit** [comment]
@scopatz It is.

**Comment by scopatz** [comment]
Thanks @nicfit!

**Pull request merged**